<a href="https://colab.research.google.com/github/santiagovillamizarjaimes-collab/Primer-corte/blob/main/quiz26marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [14]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("Ingrese el nombre del estudiante: ")

def pedir_nota():
  n1 = float(input("Nota 1: "))
  n2 = float(input("Nota 2: "))
  n3 = float(input("Nota 3: "))
  n4 = float(input("Nota 4: "))
  return [n1, n2, n3, n4]

Overwriting pedirnotas.py


In [15]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1, n2, n3, n4):
  return (n1 + n2 + n3+ n4) / 4

def mediana(n1, n2, n3, n4):
  return statistics.median([n1, n2, n3, n4])

def moda(n1, n2, n3, n4):
  return statistics.mode([n1, n2, n3, n4])

Overwriting promedio.py


In [16]:
import pedirnotas
nombre= pedirnotas.pedir_nombre()

Ingrese el nombre del estudiante: Erick Santiago


In [13]:
notas = pedirnotas.pedir_nota()

Nota 1: 5.0
Nota 2: 4.7
Nota 3: 4.7


In [17]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
notas = pedirnotas.pedir_nota()

Nota 1: 4.7
Nota 2: 4.5
Nota 3: 4.5
Nota 4: 5.0


In [28]:
import promedio
import importlib
importlib.reload(promedio)

prom = promedio.calcular_promedio(notas[0], notas[1], notas[2], notas[3])
mediana = promedio.mediana(notas[0], notas[1], notas[2], notas[3])
moda = promedio.moda(notas[0], notas[1], notas[2], notas[3])

print("Estudiante", nombre)
print("Promedio", prom)
print("Mediana:", mediana)
print("Moda:", moda)

if prom >= 3:
  print("Aprobó")
else:
  print("Reprobó")

Estudiante Erick Santiago
Promedio 4.675
Mediana: 4.6
Moda: 4.5
Aprobó



*   En promedio el estudiante obtuvo 4.675.
*   El valor que divide el 50% de las calificaciones es 4.6.
*   La calificación más repetida es 4.5 porque fue una nota obtenida en dos previos.

El estudiante obtuvo un desempeño sobresaliente, porque en promedio sus calificaciones tienen un valor superior a 4,5, es decir, que gran parte de sus calificaciones son cercanas a dicho valor.
